frontend/dockerfile/dockerfile_test.go

```diff
@@ -433,6 +434,51 @@ ADD %s /newname.tar.gz
 	require.Equal(t, buf2.Bytes(), dt)
 }
 
+func testSymlinkDestination(t *testing.T, sb integration.Sandbox) {
+	t.Parallel()
+
+	buf := bytes.NewBuffer(nil)
+	tw := tar.NewWriter(buf)
+	expectedContent := []byte("content0")
+	err := tw.WriteHeader(&tar.Header{
+		Name:     "symlink",
+		Typeflag: tar.TypeSymlink,
+		Linkname: "../tmp/symlink-target",
+		Mode:     0755,
+	})
+	require.NoError(t, err)
+	err = tw.Close()
+	require.NoError(t, err)
+
+	dockerfile := []byte(`
+FROM scratch
+ADD t.tar /
+COPY foo /symlink/
```

Comment:
could you add cases where
1) symlink doesn't end with `/`
2) symlink points to an actual file
3) broken symlink doesn't end with `/` and source is a directory